server/cluster/coordinator.go

```diff
@@ -157,6 +157,36 @@ func (c *coordinator) patrolRegions() {
 	}
 }
 
+func (c *coordinator) collectRegionStats() {
+	defer logutil.LogPanic()
+
+	defer c.wg.Done()
+	ticker := time.NewTicker(collectRegionStatsInterval)
+	defer ticker.Stop()
+	log.Info("coordinator starts collect region stats")
+	var key []byte
+	for {
+		select {
+		case <-ticker.C:
+		case <-c.ctx.Done():
+			log.Info("collect region stats has been stopped")
+			return
+		}
+
+		regions := c.cluster.ScanRegions(key, nil, statsScanRegionLimit)
+		if len(regions) == 0 {
+			// Resets the scan key.
+			key = nil
+			continue
+		}
+
+		c.cluster.updateRegionStats(regions)
+		if len(key) == 0 {
+			c.cluster.collectStatsMetrics()
```

Comment:
We can sleep a while here. If there are not many regions, we don't to collect stats too often.